src/Castle.Core/DynamicProxy/Internal/AttributeUtil.cs

```diff
@@ -144,7 +144,7 @@ public static IEnumerable<CustomAttributeInfo> GetNonInheritableAttributes(this
 							member.DeclaringType.FullName,
 							(member is TypeInfo) ? "" : ("." + member.Name),
 							typeof(AttributesToAvoidReplicating).FullName);
-					throw new ProxyGenerationException(message, e);
+					throw new DynamicProxyException(message, e);
```

Comment:
This is likely still wrong, since `DynamicProxyException` is supposedly thrown for internal errors that cannot be fixed from user code. `InvalidOperationException` might be more appropriate.

Reply:
I ended up adding a commit to make this a `NotSupportedException`.